python/usepyhf/inference.py

```diff
@@ -0,0 +1,164 @@
+import pyhf
+import numpy as np
+import pandas as pd
+import pyhf
+import numpy as np
+from usepyhf import util
+from pValue import pValueToSignificance
+
+def p_values_disc(workspace:pyhf.Workspace):
+    """
+    Hypothesis test for the discovery of a positive signal.
+    Null hypothesis: Background only
+    Alternative hypothesis: Signal+background model.
+    Discovery can be claimed if the null hypothesis can be rejected
+    at 5 sigma confidence level.
+    Uses the q0 test statistics and sets POI to 0.
+    """
+
+    #Extract model and data
+    model = workspace.model()
+    data = workspace.data(model)
+
+    #Find parameter of interest
+    poi_index = model.config.poi_index
+    poi_name = model.config.par_order[poi_index]
+
+    #Fix the parameter "SigXsec" if it exists
+    fix_pars = util.fix_sigxsec(model)
+
+    #Print information
+    print("<INFO> pyhf: Running a discovery hypothesis test:")
+    print(f"Suggested bounds for POI: { model.config.suggested_bounds()[poi_index] }")
+    print(f"Calculating p-values for POI {poi_name} = 0:")
+
+    #Calculate and print p-values
+    results = pyhf.infer.hypotest(
+            0, data, model, test_stat="q0", return_tail_probs=True,
+            fixed_params=fix_pars
+        )
+    print("")
+    print(f"CLb: {results[0]:.6f}")
+    print(f"CLs+b: {results[1][0]:.6f}")
+    print(f"CLs: {(results[1][0]/results[0]):.6f}")
+    print("")
+
+    print(f"Null hypothesis p-value: {results[0]:.6f}")
+    print(f"Significance level: {pValueToSignificance(results[0], True):.6f}")
+    print(f"Significance level needed to claim discovery: Z > 5")
+    print("")
+
+
+def p_values_excl(workspace:pyhf.Workspace):
+    """
+    Hypothesis test for the exclusion of a signal model.
+    Null hypothesis: Signal + background model
+    If the null hypothesis can be rejected at 1.65 significance
```

Comment:
I'm not sure that this is right since it's a one-sided test.  Could you please remove the reference to significance here and just stick to CLs < 0.05?  And just right CLs, I think it's not technically a p-value even though we treat it similarly.

Reply:
I added all these explanations while trying to understand how the discovery/exclusion tests work. It's the modified p-value (and the same value as CLs) so it should be clear. I think that the HistFitter output could be much more helpful. But of course it has to be correct. I will change at it, and then you can see if it looks good.

Reply:
Aright, I double checked and I misremembered.  Sorry about that.  1.65 is correct.